test/Window/Event.test.cpp

```diff
@@ -340,5 +340,29 @@ TEST_CASE("[Window] sf::Event")
             const sf::Event focusLost = sf::Event::FocusLost{};
             CHECK(focusLost.visit(visitor) == "Other");
         }
+        SECTION("Exhaustive visitor")
+        {
+            // Using a non-void return type requires providing callbacks for all types
+            CHECK(sf::Event(sf::Event::Closed{}).visit(visitor) == "Closed");
+            CHECK(sf::Event(sf::Event::Resized{}).visit(visitor) == "Resized");
+            CHECK(sf::Event(sf::Event::FocusLost{}).visit(visitor) == "Other");
+            CHECK(sf::Event(sf::Event::FocusGained{}).visit(visitor) == "Other");
+            CHECK(sf::Event(sf::Event::KeyPressed{}).visit(visitor) == "KeyPressed");
+        }
+        SECTION("Non-exhausive visitor")
+        {
+            const auto matchAll = [](auto) {};
+
+            // If you omit certain callbacks then the callbacks cannot have a return value
+            const sf::Event event = sf::Event::Closed{};
+
+            bool closedVisited = false;
+            event.visit([&closedVisited](sf::Event::Closed) { closedVisited = true; }, matchAll);
+            CHECK(closedVisited);
+
+            bool resizedVisited = false;
+            event.visit([&resizedVisited](sf::Event::Resized) { resizedVisited = true; }, matchAll);
+            CHECK(!resizedVisited);
+        }
```

Comment:
```suggestion
```
I believe the existing tests are good enough to handle these changes. Because this PR bans certain code from compiling, we don't have much to write unit tests. We don't have a way to test that certain function arguments will not compile.